playbooks/e2e.yaml

```diff
@@ -39,6 +39,16 @@
       set_fact:
         git_branch_name: "{{ git_reference | default(pull_request.json.head.ref) | default(git_reference_default) }}"
         git_repository_url: "{{ git_repo | default(pull_request.json.head.repo.clone_url) | default(git_repo_default) }}"
+    - name: Extract Zuul config
+      set_fact:
+        zuul_config: "{{ zuul.change_message | regex_search('(?s)```ZUUL_CONFIG(.+?)```', '\\1', multiline=true) }}"
```

Comment:
What will happen when `zuul.change_message` is undefined here? E.g., it is probably undefined in daily pipelines because it is not a PR.

Reply:
:thinking: good point will add default

Reply:
instead of the `default` value for `change_message` I would recommend checking whether this variable is defined, i.e. `if zuul.change_message is defined`, because this variable is not defined in the case of daily pipelines that are executed directly on the branches (main, release/v6.0.0), see e.g. [here](https://zuul.scs.community/t/SCS/build/826ad0910d4c485b821ab86e5d6367df/log/zuul-info/inventory.yaml)

Reply:
I think there still should be some default value for the `zuul_config` as it is used in the template. no? @matofeder

Reply:
agree, a default value for `zuul_config` makes sense here